test/e2e/storage/vsphere/vsphere_volume_datastore.go

```diff
@@ -67,17 +71,33 @@ var _ = utils.SIGDescribe("Volume Provisioning on Datastore [Feature:vsphere]",
 		By("Invoking Test for invalid datastore")
 		scParameters[Datastore] = InvalidDatastore
 		scParameters[DiskFormat] = ThinDisk
-		err := invokeInvalidDatastoreTestNeg(client, namespace, scParameters)
+		err := invokeDatastoreTestNeg(client, namespace, scParameters)
 		Expect(err).To(HaveOccurred())
 		errorMsg := `Failed to provision volume with StorageClass \"` + DatastoreSCName + `\": The specified datastore ` + InvalidDatastore + ` is not a shared datastore across node VMs`
 		if !strings.Contains(err.Error(), errorMsg) {
 			Expect(err).NotTo(HaveOccurred(), errorMsg)
 		}
 	})
+
+	It("verify dynamically provisioned pv using storageclass fails on a local datastore", func() {
+		By("Invoking Test for local datastore")
+		localDatastore = os.Getenv("LOCAL_DATASTORE")
+		if localDatastore == "" {
+			framework.Skipf("Environment variable LOCAL_DATASTORE is not set. Skipping the test.")
+		}
+		scParameters[Datastore] = localDatastore
+		scParameters[DiskFormat] = ThinDisk
+		err := invokeDatastoreTestNeg(client, namespace, scParameters)
+		Expect(err).To(HaveOccurred())
```

Comment:
Is there a way to check for specific error? How do we differentiate error caused by non-shared datastore from errors due to any bug in the production code or misspelled datastore name?

Reply:
I didn't find a way to check for specific error. Different errors seems to cause the same "Status" and "Reason":
    Status: Pending
    Reason: ProvisioningFailed
Only "Message" field can tell us more detailed information.